app/components/input/date-input.js

```diff
@@ -28,7 +28,8 @@ export default TimeInputComponent.extend({
 
   actions: {
     onPikadaySelection(selectedDate) {
-      this.updateDateValue(moment(selectedDate));
+      let date = moment(selectedDate);
+      this.updateDateValue(date.add(date.utcOffset(), 'minutes'));
```

Comment:
By adding the offset of the timezone to the date, we make sure that the UTC time is the same as the local time (e.g. 05-12-2022 00:00 UTC+2 becomes 05-12-2022 00:00 UTC instead of 04-12-2022 22:00 UTC)

Reply:
To me this seems like a fix to the wrong problem, but I may be wrong

We currently only do setLocale on moment in the application controller, can you try if .setTimeZone('Europe/Amsterdam') (found in ember-moment docs) fixes this?